doc/how_to_guides/stomp_planner/stomp_planner.rst

```diff
@@ -0,0 +1,109 @@
+
+Using STOMP with Your Robot
+---------------------------
+**Note:** if you are following this demo using the ``panda_moveit_config`` from the `ros-planning/panda_moveit_config <https://github.com/ros-planning/panda_moveit_config>`_ repository, these steps are already done for you and you can skip steps 1-3 and you only need to do step 4.
+
+#. Simply add the `stomp_planning.yaml <https://github.com/ros-planning/panda_moveit_config/blob/ros2/config/stomp_planning.yaml>`_ configuration file into the config directory of your MoveIt config package. It contains the plugin identifier, a planning pipeline adapter list, and the STOMP planning parameters. **Note:** The latest version of MoveIt Setup Assistant will also generate this launch file for you. The file should look like below: ::
+
+    planning_plugin: stomp_moveit/StompPlanner
+    request_adapters: >-
+      default_planner_request_adapters/AddTimeOptimalParameterization
+      default_planner_request_adapters/FixWorkspaceBounds
+      default_planner_request_adapters/FixStartStateBounds
+      default_planner_request_adapters/FixStartStateCollision
+      default_planner_request_adapters/FixStartStatePathConstraints
+
+    stomp_moveit:
+      num_timesteps: 60
+      num_iterations: 40
+      num_iterations_after_valid: 0
+      num_rollouts: 30
+      max_rollouts: 30
+      exponentiated_cost_sensitivity: 0.5
+      control_cost_weight: 0.1
+      delta_t: 0.1
+
+#. Configure MoveIt to load the STOMP planning pipeline by adding "stomp" to your MoveItConfiguration launch statement next to "ompl" and the other planners. You can find an example for this in the `demo.launch.py <https://github.com/ros-planning/moveit_resources/blob/ros2/panda_moveit_config/launch/demo.launch.py#L42>`_ of the Panda config.
+
+Running the Demo
+----------------
+If you have the ``panda_moveit_config`` from the `ros-planning/moveit_resources <https://github.com/ros-planning/moveit_resources>`_ repository you should be able to simply launch the demo setup and start planning with STOMP in RViZ ::
+
+  ros2 launch panda_moveit_config demo.launch.py
+
+STOMP Parameters
+----------------
+STOMP's parameters are configurable using the `stomp_planning.yaml <https://github.com/ros-planning/moveit_resources/blob/ros2/panda_moveit_config/config/stomp_planning.yaml>`_. All parameters are defined by the `stomp_moveit.yaml <https://github.com/ros-planning/moveit2/blob/main/moveit_planners/stomp/res/stomp_moveit.yaml>`_ which is used as input for the `generate_parameter_library <https://github.com/PickNikRobotics/generate_parameter_library>`_:
+
+**Optimization Parameters**:
+
+- *num_timesteps*: the number of timesteps used for trajectories - this number directly translates to the number of waypoints considered for trajectory initialization, during planning and for the solution output.
```

Comment:
What happens if this parameter differs from the number of waypoints of the initial trajectory? Is the reference trajectory in that case downsampled/ interpolated?

Reply:
Great question! I had to look it up. Currently the seed trajectory is directly used as input so this parameter is ignored for smoothing. We should probably fix that or document this case, especially since STOMP assumes equal time steps between the waypoints.